automation_tools/__init__.py

```diff
@@ -2127,6 +2130,151 @@ def get_hostname_from_ip(ip, timeout=3):
     return output[ip]
 
 
+def get_rhevm_client():
+    """Creates client object instance for rhevm.
+
+    Returns the client object for further use.
+
+    The following environment variables affect this command:
+
+    USERNAME
+        The username of a rhevm project to login.
+    PASSWORD
+        The password of a rhevm project to login.
+    API_URL
+        An url to API of rhevm project.
+    """
+    username = os.environ.get('USERNAME')
+    if username is None:
+        print('The USERNAME environment variable should be defined.')
+    password = os.environ.get('PASSWORD')
+    if password is None:
+        print('The PASSWORD environment variable should be defined.')
+    api_url = os.environ.get('API_URL')
+    if api_url is None:
+        print('An API_URL environment variable should be defined.')
+    try:
+        obj = API(
+            url=api_url,
+            username=username,
+            password=password,
+            insecure=True
+        )
+        return obj
+    except errors.RequestError:
+        print('ERROR ! Invalid Credentials provided for RHEVM.')
+        sys.exit(1)
+
+
+def create_rhevm_instance(instance_name, template_name, timeout=5):
+    """Creates rhevm Instance from template.
+
+    The assigning template should have network and storage configuration saved
+    already.
+
+    :param instance_name: A string. RHEVM Instance name to create.
+    :param template_name: A string. RHEVM image name from which instance
+        to be created.
+    :param int timeout: The polling timeout in minutes to create rhevm
+    instance.
+
+    ssh_key should be added to openstack project before running automation.
+    Else the automation will fail.
+
+    The following environment variables affect this command:
+
+    USERNAME
+        The username of a rhevm project to login.
+    PASSWORD
+        The password of a rhevm project to login.
+    API_URL
+        An url to API of rhevm project.
+
+    """
+    obj = get_rhevm_client()
+    temp = obj.templates.get(name=template_name)
+    dc = obj.datacenters.get(name='Default')
```

Comment:
As we are going to use the defined datacenter and quata for upgrade on rhevm, so I didnt make it optional.
But still It will be good for other feature testing in future.